packages/library/src/commonMain/kotlin/io/realm/RealmResults.kt

```diff
@@ -30,12 +31,19 @@ import kotlin.reflect.KClass
 //  - Postponing execution to actually accessing the elements also prevents query parser errors to
 //    be raised. Maybe we can get an option to prevalidate queries in the C-API?
 class RealmResults<T : RealmObject> constructor(
+    private val realmConfiguration: RealmConfiguration,
     private val realm: NativePointer,
     private val queryPointer: () -> NativePointer,
     private val clazz: KClass<T>,
     private val mediator: Mediator
 ) : AbstractList<T>(), Queryable<T> {
 
+    public var version: VersionId = VersionId(0, 0)
```

Comment:
Should we place this in a `Versioned` interface, maybe even along with the db pointer? 🤔 

Maybe questionable when naming of the property might interfere on RealmObject (#219), but we could maybe consider it when we have more insight into the outline of the different realm-pointers/versions flying around.

Reply:
Hmm, interesting idea. I think it is a bit premature given that we still don't know exactly how this looks on RealmObject nor how we pass Realms around yet, but I added it to https://github.com/realm/realm-kotlin/issues/219